platform/posix/utest/clock_utest.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "unity.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "clock.h"
+
+#include "mock_time_api.h"
+
+/* The amount of time to sleep, which is a parameter passed to #Clock_SleepMs. */
+#define SLEEP_TIME_MS                  ( 500 )
+
+/*
+ * Time conversion constants.
+ */
+#define NANOSECONDS_PER_MILLISECOND    ( 1000000L )    /**< @brief Nanoseconds per millisecond. */
+#define MILLISECONDS_PER_SECOND        ( 1000L )
+
+/**
+ * @brief Used to make assertions on the arguments passed to #nanosleep
+ * from #Clock_SleepMs.
+ *
+ * @param[in] __requested_time The requested time to sleep.
+ * @param[in] __remaining The remaining time left to sleep.
+ *
+ * @return Successful status returned by #nanosleep or 0
+ */
+int nanosleep_validate_args( const struct timespec * __requested_time,
```

Comment:
is there a reason why we use the __ added to the param names? If not, can we follow our naming conventions?